.github/workflows/publish_images.yaml

```diff
@@ -70,17 +70,29 @@ jobs:
             charmcraft release "${charm_name}" --revision=${charm_rev} --channel=edge --resource="${charm_resource}:${img_rev}"
           }
 
-          for release in $(ls $GITHUB_WORKSPACE/legend/releases/)
-                    
-          do
-            # If there was a new Legend release yesterday, release it to Charmhub
-            if [[ "$release" == $(date +%Y-%m-%d -d "yesterday") ]];          
+          previous_release=""
+          
+          # Reverting the order, so if there are multiple releases on the same month,
+          # it is published only the latest version
```

Comment:
This is a more complete comment:

```
# Filtering out releases older than 2022-04-01.
# Keep the latest release from the same month.
```

Additionally, since we're only keeping the latest release from the same month, we don't need the ``previous_release`` mechanism / if below since there won't be 2 equal release dates with this filter.

Reply:
Done!